### ISSUE TITLE
Runtime exceptions for AssertionMatcher if used with Java 8u60 and 8u65

### ISSUE THREAD

**Issue by szpak** [szpak/mockito-java8#2]
mockito-java8 stopped working with Java 8u60+ (8u65 is the latest available at the time of writing).

Errors observed in mockito-java8 build:
```
java.lang.NullPointerException
	at info.solidsoft.mockito.java8.AssertionMatcherTest.shouldAllowToUseAssertionInLambdaWithPrimitiveAsArgument(AssertionMatcherTest.java:53)
```

```
org.mockito.exceptions.misusing.UnfinishedVerificationException: 
Missing method call for verify(mock) here:
-> at info.solidsoft.mockito.java8.AssertionMatcherTest.shouldAllowToUseAssertionInLambdaWithPrimitiveAsArgument(AssertionMatcherTest.java:53)
```

```
org.mockito.exceptions.misusing.InvalidUseOfMatchersException: 
Misplaced argument matcher detected here:

-> at info.solidsoft.mockito.java8.AssertionMatcher.assertArg(AssertionMatcher.java:78)
```
It happens with Java 8u60 and 8u65 (both OracleJDK and OpenJDK). With 8u51 (and for example 8u31 on Travis) it works fine. Tested with Mockito 1.10.19 and 2.0.21-beta (mockito-java8 is [not compatible](https://github.com/szpak/mockito-java8/issues/1) with newer Mockito versions).

Full stacktraces: https://gist.github.com/szpak/f85fb33f6d27aa66ccf5.
Branch reproducing the problem: https://github.com/szpak/mockito-java8/tree/tech/strangeTestFailure
Failing Travis build with 8u66: https://travis-ci.org/szpak/mockito-java8/builds/90568711

**Comment by szpak** [comment]
Mockito 2.0.31-beta is also affected.

**Comment by szpak** [comment]
Fixed upstream in typetools [0.4.3](https://github.com/jhalterman/typetools/blob/master/CHANGES.md#bug-fixes). Released mockito-java8 0.3.1.
